WhoIsThatServer.Storage/Controllers/DatabaseImageElementController.cs

```diff
@@ -27,19 +29,61 @@ public IHttpActionResult GetAllImages()
         public IHttpActionResult Post([FromBody] DatabaseImageElement databaseImageElement)
         {
-            databaseImageElement = DatabaseImageElementHelper.InsertNewImageElement(databaseImageElement.Id,
-                databaseImageElement.ImageName, databaseImageElement.ImageContentUri,
-                databaseImageElement.PersonFirstName, databaseImageElement.PersonLastName, databaseImageElement.DescriptiveSentence, databaseImageElement.Score);
+            try
+            {
+                databaseImageElement = DatabaseImageElementHelper.InsertNewImageElement(databaseImageElement.Id,
+                    databaseImageElement.ImageName, databaseImageElement.ImageContentUri,
+                    databaseImageElement.PersonFirstName, databaseImageElement.PersonLastName,
+                    databaseImageElement.DescriptiveSentence, databaseImageElement.Score);
 
-            return Json(databaseImageElement);
+                return Json(databaseImageElement);
+            }
+
+            catch (ManagerException wrongUriException) when (wrongUriException.ErrorCode ==
+                                                             StorageErrorMessages.InvalidImageUriError)
+            {
+                return Json(StorageErrorMessages.InvalidImageUriError);
+            }
+
+            catch (ManagerException wrongFilenameException) when (wrongFilenameException.ErrorCode ==
+                                                                  StorageErrorMessages.InvalidFileNameError)
+            {
+                return Json(StorageErrorMessages.InvalidFileNameError);
+            }
         }
 
         /// <inheritdoc/>
         [HttpPut]
         [Route("api/images/score")]
         public IHttpActionResult UpdateScore([FromBody] DatabaseImageElement databaseImageElement)
         {
-            return Json(DatabaseImageElementHelper.UpdateScore(databaseImageElement.Id));
+            try
+            {
+                return Json(DatabaseImageElementHelper.UpdateScore(databaseImageElement.Id));
+            }
+
+            catch (ManagerException userNotFoundException) when (userNotFoundException.ErrorCode ==
+                                                                 StorageErrorMessages.UserDoesNotExistError)
+            {
+                return Json(StorageErrorMessages.UserDoesNotExistError);
+            }
+        }
+
+        /// <inheritdoc/>
+        [HttpGet]
+        [Route("api/images/user/{id}")]
+        public IHttpActionResult GetUserById(int id)
+        {
+            try
+            {
+                return Json(DatabaseImageElementHelper.GetUserById(id));
+            }
+
+            catch (ManagerException userNotFoundException) when (userNotFoundException.ErrorCode ==
+                                                                 StorageErrorMessages.UserDoesNotExistError)
+            {
+                return Json(StorageErrorMessages.UserDoesNotExistError);
```

Comment:
One small notice - you can return the object straight away by changing `IHttpActionResult` to `YourClass`. It will be serialized automatically.
Other than that, good!